contributing/development/compiling/compiling_for_linuxbsd.rst

```diff
@@ -363,6 +363,63 @@ You don't even need to copy them, you can just reference the resulting
 files in the ``bin/`` directory of your Godot source folder, so the next
 time you build, you automatically have the custom templates referenced.
 
+Cross-compiling for RISC-V devices
+----------------------------------
+
+To cross-compile Godot for RISC-V devices, we need to setup the following items:
+
+- `riscv-gnu-toolchain <https://github.com/riscv-collab/riscv-gnu-toolchain/releases>`__.
+  While we are not going to use this directly, it provides us with a sysroot, as well
+  as header and libraries files that we will need. There are many versions to choose
+  from, however, the older the toolchain, the more compatible our final binaries will be.
+  If in doubt, `use this version <https://github.com/riscv-collab/riscv-gnu-toolchain/releases/tag/2021.12.22>`__,
+  and download ``riscv64-glibc-ubuntu-18.04-nightly-2021.12.22-nightly.tar.gz``. Extract
+  it somewhere and remember its path.
+- Clang. RISC-V GCC has
+  `bugs with its atomic operations <https://github.com/riscv-collab/riscv-gcc/issues/15>`__
+  which prevent it from compiling Godot correctly. Any version of Clang from 16.0.0 upwards
+  will suffice. Download it from the package manager of your distro, and make sure that
+  it *can* compile to RISC-V. You can verify by executing this command ``clang -print-targets``,
+  make sure you see ``riscv64`` on the list of targets.
+- `mold <https://github.com/rui314/mold/releases>`__. This fast linker,
+  is the only one that correctly links the resulting binary. Download it, extract it,
+  and make sure to add its ``bin`` folder to your PATH.
+
+With all the above setup, we are now ready to build Godot.
+
+Go to the root of the source code, and execute the following build command:
+
+::
+
+    scons arch="rv64" use_llvm="yes" linker="mold" lto="none" target="editor" \
+        ccflags="--sysroot=<GCC-TOOLCHAIN-PATH>/sysroot --gcc-toolchain=<GCC-TOOLCHAIN-PATH> -target riscv64-unknown-linux-gnu" \
+        linkflags="--sysroot=<GCC-TOOLCHAIN-PATH>/sysroot --gcc-toolchain=<GCC-TOOLCHAIN-PATH> -target riscv64-unknown-linux-gnu"
```

Comment:
The quotes aren't actually needed for the first line of arguments, and we don't use them for other parts of the documentation, including other sections on the same page.

Instead of `<GCC-TOOLCHAIN-PATH>`, I believe we should encourage users to make it an environment variable, such as telling the user to put `export RISCV_TOOLCHAIN_PATH="your path here"` in your profile/aliases/whatever.

```suggestion
scons arch=rv64 use_llvm=yes linker=mold lto=none target=editor \
    ccflags="--sysroot=$RISCV_TOOLCHAIN_PATH/sysroot --gcc-toolchain=$RISCV_TOOLCHAIN_PATH -target riscv64-unknown-linux-gnu" \
    linkflags="--sysroot=$RISCV_TOOLCHAIN_PATH/sysroot --gcc-toolchain=$RISCV_TOOLCHAIN_PATH -target riscv64-unknown-linux-gnu"
```